### PR TITLE
First time, project wizard, settings viewmodel updates

### PR DESCRIPTION
**Pull request by kote2ster**
First time, project wizard, settings viewmodel updates

Implemented:
- Add custom handycontrol error handling style for input fields
- Add logic for input field validation on wizard windows
- Move settings vm to first time setup wizard vm
- Add first time setup wizard window
- Add create project window
- Add exe/dir validation in settings

Fixed:
- Fix settings json saving and loading

### CODE REVIEW

File: WolvenKit/Model/Wizards/FirstSetupWizardModel.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            get => _email;
+            set
+            {
+                _email = value;
+                RaisePropertyChanged(nameof(Email));
+            }
+        }
+
+        /// <summary>
+		/// Gets/Sets the donate link.
```
Comment (Traderain) [inline WolvenKit/Model/Wizards/FirstSetupWizardModel.cs:75]: Comments formatting

File: WolvenKit/Services/ApplicationInitializationService.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ public override async Task InitializeAfterShowingShellAsync()
             var settings = ServiceLocator.Default.ResolveType<ISettingsManager>();
             if (!File.Exists(settings.W3ExecutablePath) || !File.Exists(settings.WccLitePath))
             {
-                _commandManager.GetCommand(AppCommands.Settings.General).SafeExecute();
+                //_commandManager.GetCommand(AppCommands.Settings.General).SafeExecute();
```
Comment (Traderain) [inline WolvenKit/Services/ApplicationInitializationService.cs:80]: Why is this commented out?

File: WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs [hunk 3]
```diff
@@ -1,13 +1,464 @@
...
+
+            // automatically scan the registry for exe paths for wcc and tw3
+            // if either text field is empty
+            if (string.IsNullOrEmpty(W3ExePath) || string.IsNullOrEmpty(WccLitePath) || string.IsNullOrEmpty(CP77ExePath))
+            {
+                exeSearcherSlave_DoWork();
+            }
+
+            //TODO: handle this case!
+            if (!TryCopyOodleLib())
```
Comment (Traderain) [inline WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs:76]: Handle the error here properly, maybe filenotfoundexception or something

File: WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs [hunk 4]
```diff
@@ -1,13 +1,464 @@
...
+            _settingsManager.WccLitePath = WccLitePath;
+            //_settingsManager.GameDlcDir = GameDlcDir;
+            //_settingsManager.GameModDir = GameModDir;
+
+            _settingsManager.Save();
+
+            return await base.SaveAsync();
+        }
+
+        //private void SetDefaultModDir()
```
Comment (Traderain) [inline WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs:294]: Delete these

File: WolvenKit/Views/Wizards/WizardPages/FirstSetupWizard/FinalizeSetupView.xaml.cs [hunk 5]
```diff
@@ -1,11 +1,72 @@
...
+                _fswm.ProfileImageBrushPath = imgSelector.Uri.AbsoluteUri;
+            }
+            else
+            {
+                _fswm.ProfileImageBrush = default(System.Windows.Media.ImageBrush);
+                _fswm.ProfileImageBrushPath = "";
+            }
+        }
+
+        private async void ConfirmSettings_Click(object sender, System.Windows.RoutedEventArgs e)
```
Comment (Traderain) [inline WolvenKit/Views/Wizards/WizardPages/FirstSetupWizard/FinalizeSetupView.xaml.cs:60]: Copy the picture to the project dir

File: WolvenKit/Views/Wizards/WizardPages/ProjectWizard/ProjectConfigurationView.xaml.cs [hunk 6]
```diff
@@ -1,19 +1,59 @@
...
+            else
+            {
+                imgSelector.SetValue(ImageSelector.UriPropertyKey, default(System.Uri));
+                imgSelector.SetValue(ImageSelector.PreviewBrushPropertyKey, default(System.Windows.Media.Brush));
+                imgSelector.SetValue(ImageSelector.HasValuePropertyKey, false);
+                imgSelector.SetCurrentValue(ImageSelector.ToolTipProperty, default);
+            }
+        }
+
+        private void imgSelector_Executed(object sender, System.Windows.Input.ExecutedRoutedEventArgs e)
```
Comment (Traderain) [inline WolvenKit/Views/Wizards/WizardPages/ProjectWizard/ProjectConfigurationView.xaml.cs:44]: Copy the image

**Review by Traderain — CHANGES_REQUESTED** [review]
Please use more comments in the wizard classes, an xmldoc on the top is enough but at least people know what happens there. Otherwise <3

**Comment by Traderain** [comment]
@rfuzzo @Offline-R503B Can you guys test out the behaviour?

File: WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs (hunk 4 above)
Comment (rfuzzo) [reply WolvenKit/ViewModels/Wizards/FirstSetupWizardViewModel.cs:294]: I can handle this in my PR

**Review by Traderain — APPROVED** [review]

**Pull request merged**